openpype/modules/sync_server/sync_server.py

```diff
@@ -189,6 +205,166 @@ def _site_is_working(module, project_name, site_name, site_config):
+    if not sync_server or not sync_server.enabled:
+        print("Sync server module is disabled or unavailable.")
+        return
+
+    if subset_id is None:
+        print(
+            "Not any matched subset for task '{}' of '{}'.".format(
+                task_name, asset_name
+            )
+        )
+        return
+
+    if not workfile_representation:
+        print(
+            "Not published workfile for task '{}' and host '{}'.".format(
+                task_name, host_name
+            )
+        )
+        return
+
+    # If representation isn't available on remote site, then return.
+    if not sync_server.is_representation_on_site(
+        project_name,
+        workfile_representation["_id"],
+        sync_server.get_remote_site(project_name),
+    ):
+        print(
+            "Representation for task '{}' and host '{}'".format(
+                task_name, host_name
+            )
+        )
+        return
+
+    # Get local site
+    local_site_id = get_local_site_id()
+
+    # Add workfile representation to local site
+    representation_ids = {workfile_representation["_id"]}
+    representation_ids.update(
+        get_linked_representation_id(
+            project_name, repre_id=workfile_representation["_id"]
+        )
+    )
+    for repre_id in representation_ids:
+        if not sync_server.is_representation_on_site(project_name, repre_id,
+                                                     local_site_id):
+            sync_server.add_site(
+                project_name,
+                repre_id,
+                local_site_id,
+                force=True,
+                priority=99
+            )
+    sync_server.reset_timer()
+
+    # While representation unavailable locally, wait.
+    while not sync_server.is_representation_on_site(
+        project_name, workfile_representation["_id"], local_site_id
+    ):
+        sleep(5)
+
+    if not last_published_workfile_path:
+        last_published_workfile_path = get_representation_path_with_anatomy(
+            workfile_representation, anatomy
+        )
+    project_doc = get_project(project_name)
+
+    # Get workfile data
+    workfile_data = get_template_data(
+        project_doc, asset_doc, task_name, host_name
+    )
+
+    extension = last_published_workfile_path.split(".")[-1]
+
+    project_settings = get_project_settings(project_name)
+    template_key = get_workfile_template_key(
+        task_name, host_name, project_name, project_settings
+    )
+
+    # Get local workfile version number
+    _, last_local_workfile_version = get_last_workfile_with_version(
+        get_workdir(
+            project_doc,
+            asset_doc,
+            task_name,
+            host_name,
+            anatomy=anatomy,
+            template_key=template_key,
+            project_settings=project_settings,
+        ),
+        anatomy.templates[template_key]["file"],
+        workfile_data,
+        [extension],
+    )
```

Comment:
When does this happen through the prelaunch hook? The prelaunch hook only continues download if there's no local workfiles yet, right? So why does it try incrementing against local workfiles? Or what am I missing?

Reply:
Hm, I think this might not be necessary, removed it.
Maybe @Tilix4 will have an idea why it is there.

Reply:
I'm not sure the highlighted code is relevant. But the idea is that it takes the last version number and increments. This way, you don't get confused if you're downloading the `v003` for the first time, the opened workfile is a `v004`, and not a `v001`.

Reply:
> I'm not sure the highlighted code is relevant. But the idea is that it takes the last version number and increments. This way, you don't get confused if you're downloading the v003 for the first time, the opened workfile is a v004, and not a v001.

Yes, that's what I'd expect too - but the logic that was here originally compared it with versions on disk locally which doesn't trigger since the logic wouldn't start if there were local files.